trading_cost.py

```diff
@@ -0,0 +1,57 @@
+"""
+A file to build cost objects.
+"""
+
+class CurrencyCost:
+    """
+    Cost function of currency trading.
+    """
+    def __init__(self, base_ins, instruments):
+        self.base_ins = base_ins
+        self.cost = 0
+        self.profit = 0
+        self.quantity = {
+            instrument : 0 for instrument in instruments}
+        self.value = {
+            instrument : 0 for instrument in instruments}
+        self.value[base_ins] = 1
+        self.quantity[base_ins] = 1
+
+    def __call__(self, portfolio, price):
+        self.reset()
+        self.calculate_profit(portfolio[-1], price)
```

Comment:
Due to possibility of changing the profit-calculating way, so I separate it as another method to make it easier to read and change.

Reply:
As this function will return not only pnl but also trading cost, I suggest to call it `calculate` instead of `calculate_profit`, which may lead to misunderstanding.